eng/clipack/Common.projitems

```diff
@@ -0,0 +1,59 @@
+<Project>
+  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />
+
+  <PropertyGroup>
+    <TargetFramework>net9.0</TargetFramework>
+    <_PublishOutputDirectory>$(ArtifactsBinDir)Aspire.Cli/$(Configuration)/net8.0/$(CliRuntime)/publish/</_PublishOutputDirectory>
+  </PropertyGroup>
+
+  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
+
+  <Target Name="Build" />
+
+  <Target Name="BeforeBuild" BeforeTargets="Build">
+    <ItemGroup>
+      <AdditionalProperties Include="Configuration=$(Configuration)" />
+      <AdditionalProperties Include="RuntimeIdentifier=$(CliRuntime)" />
+      <AdditionalProperties Include="PublishSingleFile=true" />
+      <AdditionalProperties Include="SelfContained=true" />
+    </ItemGroup>
+
+    <MSBuild Projects="../../src/Aspire.Cli/Aspire.Cli.csproj" Targets="publish" Properties="@(AdditionalProperties)" />
+
+    <ItemGroup>
+      <_PublishItems Include="$(_PublishOutputDirectory)**/*" />
+    </ItemGroup>
+
+    <!-- Throw an error if _PublishItems is empty. -->
+    <Error Condition="'@(_PublishItems)' == ''"
+      Text="No files were found to pack in $(_PublishOutputDirectory). Ensure that the project $(MSBuildProjectFullPath) has a publish target defined." />
+
+    <MakeDir Directories="$(CliPublishedArtifactsOutputDir)/$(CliRuntime)" />
+
+    <PropertyGroup>
+      <_OutputArchiveDir>$([MSBuild]::NormalizeDirectory($(CliPublishedArtifactsOutputDir), $(CliRuntime)))</_OutputArchiveDir>
+      <_OutputFilenameWithoutExtension>$(_OutputArchiveDir)aspire-cli-$(CliRuntime)</_OutputFilenameWithoutExtension>
+      <_CreateTarGzScriptPath>$(RepositoryEngineeringDir)scripts\create-tar-gz.ps1</_CreateTarGzScriptPath>
+    </PropertyGroup>
+
+    <!-- Create zip for non-Linux runtimes -->
+    <ZipDirectory
+      Condition="!$(CliRuntime.StartsWith('linux-'))"
+      SourceDirectory="$(_PublishOutputDirectory)"
+      DestinationFile="$(_OutputFilenameWithoutExtension).zip"
+      Overwrite="true" />
+
+    <!-- Create tar.gz for Linux runtimes using PowerShell script -->
+    <ItemGroup>
+      <_CreateTarGzArguments Include="-SourceDirectory &quot;$(_PublishOutputDirectory)&quot;" />
+      <_CreateTarGzArguments Include="-DestinationFile &quot;$(_OutputFilenameWithoutExtension).tar.gz&quot;" />
+      <_CreateTarGzArguments Include="-Overwrite true" />
+    </ItemGroup>
+
+    <Exec
+      Condition="$(CliRuntime.StartsWith('linux-'))"
+      Command="pwsh &quot;$(_CreateTarGzScriptPath)&quot; @(_CreateTarGzArguments, ' ')"
```

Comment:
This now means there is a new prereq in the build as I don't believe pwsh was required before, correct?

Was there nothing in the SDK that does this naitvely? Might also be worth checking with dotnet monitor CLI tool as they produce tars as part of their build so we should also see what they do. TL:DR is that if we can simplify and get to the same results without having to introduce new pre-reqs and custom scripts, it'd be better.

Reply:
re:pwsh, we use that in multiple places for running tests on github actions.

re:sdk, there is a https://github.com/dotnet/sdk/blob/ad24473f7f2ae16eaee4cbd28eb6897d00df32f3/src/Tasks/sdk-tasks/TarGzFileCreateFromDirectory.cs#L8 which isn't being surfaced in a nuget. But Eric suggested https://github.com/dotnet/arcade/tree/main/src/Microsoft.DotNet.Tar which I will be trying. Still in draft :D

Reply:
Excellent find! There seems to be `[Microsoft.DotNet.Build.Tasks.Archives](https://github.com/dotnet/arcade/tree/d9d02d858b71562509f72ea84409853e4deffc8f/src/Microsoft.DotNet.Build.Tasks.Archives)` nuget which IIUC supports this. I will try it out.